src/models/index.js

```diff
@@ -24,6 +24,51 @@ if (config.use_env_variable) {
 
 audit.attachHooksForAuditing(sequelize);
 
+// Function to gracefully close the Sequelize connection
+const gracefulShutdown = async (msg) => {
+  try {
+    await sequelize.close();
+    auditLogger.info(`Sequelize disconnected through ${msg}`);
+  } catch (err) {
+    auditLogger.error(`Error during Sequelize disconnection through ${msg}: ${err}`);
+  }
+};
+
+// Listen for SIGINT (Ctrl+C)
+process.on('SIGINT', async () => {
+  await gracefulShutdown('app termination (SIGINT)');
+  process.exit(0);
+});
+
+// Listen for SIGTERM (e.g., kill command)
+process.on('SIGTERM', async () => {
+  await gracefulShutdown('app termination (SIGTERM)');
+  process.exit(0);
+});
+
+// Listen for uncaught exceptions
+process.on('uncaughtException', async (err) => {
+  auditLogger.error('Uncaught Exception:', err);
+  await gracefulShutdown('uncaught exception');
+  process.exit(1);
+});
+
+sequelize.addHook('beforeConnect', () => {
+  auditLogger.info('Attempting to connect to the database');
```

Comment:
Is it at all possible to give more context to what is attempting to create a connection to the database. IE a specific model or function? Seems to work fine I just wonder if the next question would be what is attempting this operation.

Reply:
I can understand why that might be helpful, but there is no real method to know or pass anything to this function. As the  use/reuse of connections within the pool is fully handled by sequelize. We don't have a good way of knowing when it will create a new connection. For example when running tests that access the database, because they are not in a transaction, every query seems to be using a separate connection. But when used in a more traditional path, one connection is used for the scope of the transaction. 

That said, i might be able to get the some information in. Since all calls via a route come through the transaction wrapper, and soon all actions done by workers will go through a similar transaction wrapper. I could add some logic to read something from the httpContext and add that to the comment posted in the info message.

Items that might be available are:

- loggedUser
- impersonatedUserId
- sessionSig
- transactionId - the one used by the audit log
- function used by the route

note that some calls to the DB occur before these are defined, so they may not be available for all connections.

What would the proposed message look like for each of the (dis)connect logs?

Reply:
On the connections from the worker, I am currently trying to get/understand how to do a similar thing as currently none of the vars below are set:

- loggedUser
- impersonatedUserId
- sessionSig
- transactionId - the one used by the audit log

which will result in the audit log being less useful